addon/utils/proxy-unproxy-objects.js

```diff
@@ -0,0 +1,46 @@
+import { isNone } from '@ember/utils';
+/**
+ * This utility is ussed to create a wrapper object around the object passed to the proxy function.
+ * Currently on drag action, the `draggable-object` mutates the `content` object and appends a new
+ * property `isDraggingObject` onto the parent object.
+ * This behavior throws an exception with M3 models because it not a previously whitelisted property. The issue
+ * is being tracked (ticket: TODO: Add Github issue ticket here).
+ *
+ * This utility is used to create a proxy object to wrap the original `content` M3 objet to 
+ * avoud any exceptions from being thrown when dragging is initiated on the object.
+ * 
+ * Note: Added further implementation to `unproxyObject` that can be used to unproxy the previously
+ * proxies object thereby returning the original M3 object.
+ *
+ */
+
+/**
+ * @access public
+ * Returns the passed object wrapped within new object.
+ * Used to proxy draggable objects.
+ * @param objectToProxy Object to proxy.
+ * @returns {Object} Proxy object.
+ */
+export function proxyObject(objectToProxy) {
+  if (!isNone(objectToProxy)) {
+    return {
+      content: objectToProxy,
+      id: objectToProxy.id,
```

Comment:
we may want to make "key" configurable for the more general case. One option might be to proxy `guidFor` from the target.

Reply:
I just rolled out an update where im using `guid`, but as you can see, I still need a way to somehow persist the name of the key under which I need to save the object. Originally it was hard coded to `content`, whereas now its more dynamic. Still need something like `unwrappingKey` that stores the wrapping key so that we can unwrap it correctly. Please let me know if this approach is unclear. We can further have a discussion about this offline. For now i'll go ahead and resolve the issue.